timesketch/lib/sigma_util.py

```diff
@@ -279,14 +283,13 @@ def get_sigma_rule_by_text(rule_text, sigma_config=None):
     sigma_backend = sigma_es.ElasticsearchQuerystringBackend(sigma_conf_obj, {})
 
     rule_return = {}
-
     # TODO check if input validation is needed / useful.
     try:
-        parser = sigma_collection.SigmaCollectionParser(
-            rule_text, sigma_conf_obj, None)
-        parsed_sigma_rules = parser.generate(sigma_backend)
         rule_yaml_data = yaml.safe_load_all(rule_text)
         for doc in rule_yaml_data:
+            parser = sigma_collection.SigmaCollectionParser(
+                str(doc), sigma_conf_obj, None)
```

Comment:
why the `str` casting?

Reply:
wait, so is this sometimes a dict and sometimes a string?

I fell like that should be handled better than just by string casting

Reply:
the upstream function does this:
```
self.yamls = yaml.safe_load_all(content)
```

So it expects this to be a string, more specifically a YAML string.... so if we have it as a dict, we should dump it as a YAML string, rather than string cast it